### PR TITLE
Feature: first commit

### PR DESCRIPTION
**Pull request by SamSalvatico**
Hello everyone!

This is the first commit of the `shared-node-utils` repository.

I've added the `fastify-logging-wrapper` and the `fastify-error-handler` packages.

### CODE REVIEW

File: commitlint.config.js [hunk 1]
```diff
@@ -1,3 +1,15 @@
 module.exports = {
   extends: ["@commitlint/config-conventional"],
+  rules: {
+    "scope-enum": [
+      2,
+      "always",
+      [
+        "core",
```
Comment (alewin) [inline commitlint.config.js:8]: is `core` scope necessary? what would be its use case? or is it just a typo?

**Review by alewin — COMMENTED** [review]
there are some files without EOF line

File: packages/fastify-error-handler/package.json [hunk 2]
```diff
@@ -5,14 +5,23 @@
...
+    "description": "Normalize the error handling of errors with related logs",
     "dependencies": {
-        "@ogcio/shared-errors": "^0.0.1"
+        "@fastify/sensible": "^5.6.0",
+        "@ogcio/fastify-logging-wrapper": "^0.0.1",
+        "@ogcio/shared-errors": "^0.0.1",
+        "fastify": "^4.28.1"
+    },
+    "devDependencies": {
+        "@types/http-errors": "^2.0.4"
```
Comment (alewin) [inline packages/fastify-error-handler/package.json:25]: why `@types/http-errors` is needed? we are not using it, in `fastify-error-handler`

File: commitlint.config.js (hunk 1 above)
Comment (SamSalvatico) [reply commitlint.config.js:8]: I think it is, it's meant to be used for updates related to the root repository, like for this file 😁

File: packages/fastify-error-handler/package.json (hunk 2 above)
Comment (SamSalvatico) [reply packages/fastify-error-handler/package.json:25]: Yes, we are
https://github.com/ogcio/shared-node-utils/blob/70487eb0e9ad00c3064d18ae59b9526994fa5d8f/packages/fastify-error-handler/src/initialize-error-handler.ts#L18

File: commitlint.config.js (hunk 1 above)
Comment (alewin) [reply commitlint.config.js:8]: okok, clear thanks! 👌 
alternatively I guess we could also omit the scope

File: packages/fastify-error-handler/src/initialize-error-handler.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  if (error.validation && error.validation.length > 0) {
+    output.validation = getValidationFromFastifyError(
+      error.validation
+    ).validation;
+  }
+
+  return output;
+};
+
+const manageLifeEventsError = (
```
Comment (alewin) [inline packages/fastify-error-handler/src/initialize-error-handler.ts:141]: we could omit the prefix ‘lifeEvents’

Comment (SamSalvatico) [reply packages/fastify-error-handler/src/initialize-error-handler.ts]: thank you, @alewin

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
 node_modules
 *.tsbuildinfo
-dist
\ No newline at end of file
+dist
+.tap
```
Comment (andreacappadona17) [inline .gitignore:4]: EOF

File: lerna.json [hunk 5]
```diff
@@ -3,5 +3,10 @@
...
     "packages/*"
   ],
-  "version": "independent"
+  "version": "independent",
+  "command": {
+    "version": {
+      "conventionalCommits": true
+    }
+  }
 }
```
Comment (andreacappadona17) [inline lerna.json:12]: EOF

File: packages/fastify-logging-wrapper/package.json [hunk 6]
```diff
@@ -5,14 +5,24 @@
...
+        "@fastify/sensible": "^5.6.0",
+        "@ogcio/shared-errors": "^0.0.1",
+        "fastify": "^4.28.1",
+        "hyperid": "^3.3.0"
+    },
+    "devDependencies": {
+        "@types/http-errors": "^2.0.4"
     }
-}
+}
```
Comment (andreacappadona17) [inline packages/fastify-logging-wrapper/package.json:28]: EOF

File: packages/shared-errors/package.json [hunk 7]
```diff
@@ -5,11 +5,16 @@
...
-    "description": ""
-}
+    "description": "",
+    "dependencies": {
+        "http-errors": "^2.0.0"
+    },
+    "devDependencies": {
+        "@types/http-errors": "^2.0.4"
+    }
+}
```
Comment (andreacappadona17) [inline packages/shared-errors/package.json:20]: EOF

File: packages/shared-errors/tsconfig.json [hunk 8]
```diff
@@ -3,7 +3,7 @@
     "compilerOptions": {
       "composite": true,
       "outDir": "dist",
-      "rootDir": "src"
+      "rootDir": "src",
     },
     "include": ["src"]
   }
```
Comment (andreacappadona17) [inline packages/shared-errors/tsconfig.json:9]: EOF

File: tsconfig.base.json [hunk 9]
```diff
@@ -1,13 +1,14 @@
...
     "skipLibCheck": true,
     "forceConsistentCasingInFileNames": true,
     "declaration": true,
     "declarationMap": true,
     "sourceMap": true,
-    "module": "ES2020"
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext"
   }
 }
```
Comment (andreacappadona17) [inline tsconfig.base.json:14]: EOF

**Review by andreacappadona17 — COMMENTED** [review]
Sorry for doing EOF police 👮‍♂️

**Comment by SamSalvatico** [comment]
> Sorry for doing EOF police 👮‍♂️

EOF police is always welcomed

**Review by alewin — APPROVED** [review]

**Review by andreacappadona17 — APPROVED** [review]
LGTM

**Pull request merged**